src/configure/auto_mozilla.rs

```diff
@@ -35,25 +37,20 @@ enum MozConfigTag {
     Username,
 }
 
-pub fn moz_autoconfigure(
-    context: &Context,
-    url: &str,
-    param_in: &LoginParam,
-) -> Option<LoginParam> {
-    let xml_raw = read_autoconf_file(context, url)?;
+pub fn moz_parse_xml(in_emailaddr: &str, xml_raw: &str) -> Result<LoginParam, Error> {
+    let mut reader = quick_xml::Reader::from_str(xml_raw);
+    reader.trim_text(true);
 
     // Split address into local part and domain part.
-    let p = param_in.addr.find('@')?;
-    let (in_emaillocalpart, in_emaildomain) = param_in.addr.split_at(p);
+    let p = match in_emailaddr.find('@') {
+        Some(i) => i,
+        None => bail!("Email address {} does not contain @", in_emailaddr),
+    };
+    let (in_emaillocalpart, in_emaildomain) = in_emailaddr.split_at(p);
```

Comment:
Does it make sense to use `dc_tools::EmailAddress` here?

Reply:
EmailAddress stores owned strings, so it is overkill here. If parsing and verification are factored out, then maybe, but two `to_string` that it does at the end are not necessary.